### PR TITLE
add pgvector support

### PR DESCRIPTION
**Pull request by dusvyat**
## Description

PR to add support for pgvector

**Fixes** #(issue)

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📄 This change requires a documentation update

### What is the solution?

install pgvector  extension  in postgres (11+) 

```bash
cd /tmp
git clone --branch v0.4.4 https://github.com/pgvector/pgvector.git
cd pgvector
make
make install
```

Create a pgvector handler

```sql
CREATE DATABASE pvec
WITH
    ENGINE = 'pgvector',
    PARAMETERS = {
    "host": "127.0.0.1",
    "port": 5432,
    "database": "postgres",
    "user": "d",
    "password": "password"
    };
```

Add embeddings to a pgvector collection

```sql
CREATE TABLE pvec.items
    (select embedding from (select * from mysql_demo_db.demo_fda_context limit 3) as d
join  openai_emb);
```

create a search embedding (one that should def return a result)

```sql
CREATE VIEW mindsdb.embedding as (
(select embedding from (select * from mysql_demo_db.demo_fda_context limit 1) as d
join  openai_emb)
);
```

Do a search from an input search embedding

```sql
SELECT * FROM pvec.items where embedding = (select * from mindsdb.embedding) limit 10;
```

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation, or created issues to update them.
- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] I have shared a short loom video or screenshots demonstrating any new functionality.

### CODE REVIEW

File: mindsdb/integrations/handlers/pgvector_handler/pgvector_handler.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        collection_name = query.from_table.parts[-1]
+
+        with self.connection.cursor() as cur:
+            try:
+                # convert search embedding to string
+                string_embeddings_search = str(query.where.args[1].items[0].value)
+                # get limit from query
+                limit = query.limit.value if query.limit else 5
+                # we need to use the <-> operator to search for similar vectors,
+                # so we need to convert the string to a vector and also use a threshold (e.g. 0.5)
```
Comment (paxcema) [inline mindsdb/integrations/handlers/pgvector_handler/pgvector_handler.py:83]: Shouldn't this threshold be configurable through `using`?

**Review by paxcema — APPROVED** [review]
Generally looks good to me. I've looped @ZoranPandovski in so he can take a look on the rest of the handler, although I don't spot anything wrong with that either.

File: mindsdb/integrations/handlers/chromadb_handler/chromadb_handler.py [hunk 2]
```diff
@@ -279,6 +279,10 @@ def insert(
 
         collection = self._client.get_collection(table_name)
 
+        # drop columns with all None values
+
+        data.dropna(axis=1, inplace=True)
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/chromadb_handler/chromadb_handler.py:284]: move out of base class into chroma, as its chroma specific for now

File: mindsdb/integrations/libs/vectordatabase_handler.py [hunk 3]
```diff
@@ -262,10 +262,6 @@ def _dispatch_insert(self, query: Insert) -> HandlerResponse:
             }
         )
 
-        # drop columns with all None values
-
-        data.dropna(axis=1, inplace=True)
-
```
Comment (dusvyat) [inline mindsdb/integrations/libs/vectordatabase_handler.py:268]: moved into chroma handler

File: mindsdb/integrations/handlers/pgvector_handler/README.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
+# PGVector Handler
+
+This is the implementation of the PGVector for MindsDB.
+
+## PGVector
+
+Chroma is the open-source embedding database. Chroma makes it easy to build LLM apps by making knowledge, facts, and skills pluggable for LLMs.
```
[the remark this answers is not in the document]
Comment (dusvyat) [reply mindsdb/integrations/handlers/pgvector_handler/README.md:7]: my bad, will fix

File: mindsdb/integrations/handlers/pgvector_handler/pgvector_handler.py [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+    FilterCondition,
+    FilterOperator,
+    TableField,
+    VectorStoreHandler,
+)
+from mindsdb.utilities import log
+from mindsdb.utilities.profiler import profiler
+
+
+# todo add support for different indexes and search algorithms e.g. cosine similarity or L2 norm
```
[the remark this answers is not in the document]
Comment (dusvyat) [reply mindsdb/integrations/handlers/pgvector_handler/pgvector_handler.py:27]: good point will add issue

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**